docs/ceph-csi-upgrade.md

```diff
@@ -0,0 +1,334 @@
+updated, you need to be aware of whether this affects your applications. This
+issue will happen when using the Ceph fuse client or rbd-nbd:
+
+CephFS: If you are provision volumes for CephFS and have a kernel less than
+version 4.17, The CSI driver will fall back to use the FUSE client.  RBD: If
+you have set the mounter: rbd-nbd option in the RBD storage class, the NBD
+mounter will have this issue.
+
+If you are affected by this issue, you will need to proceed carefully during
+the upgrade to restart your application pods. The recommended step is to modify
+the update strategy of the CSI nodeplugin daemonsets to OnDelete so that you
+can control when the CSI driver pods are restarted on each node.
+
+To avoid this issue in future upgrades, we recommend that you do not use the
+fuse client or rbd-nbd as of now.
+
+#### 3. Upgrade CephFS Nodeplugin resources
+
+upgrading nodeplugin resources include updating the nodeplugin RBAC and
+nodeplugin daemonset
+
+##### 3.1 Update the CephFS Nodeplugin RBAC
+
+```bash
+$]kubectl apply -f deploy/cephfs/kubernetes/v1.14+/csi-nodeplugin-rbac.yaml
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+serviceaccount/cephfs-csi-nodeplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-csi-nodeplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-csi-nodeplugin-rules configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrolebinding.rbac.authorization.k8s.io/cephfs-csi-nodeplugin configured
+```
+
+**Note:** Ignore the warning messages
+
+If you determined in step 2 that you were affected by the CSI driver restart
+issue that disconnects the application pods from their mounts, continue with
+this section. Otherwise, you can skip to step 3.2
+
+```console
+vi deploy/cephfs/kubernetes/v1.14+/csi-cephfsplugin.yaml
+```
+
+```yaml
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: csi-cephfsplugin
+spec:
+  selector:
+    matchLabels:
+      app: csi-cephfsplugin
+  updateStrategy:
+    type: OnDelete
+  template:
+    metadata:
+      labels:
+        app: csi-cephfsplugin
+    spec:
+      serviceAccount: cephfs-csi-nodeplugin
+```
+
+##### 3.2 Update the CephFS Nodeplugin daemonset
+
+```bash
+[$]kubectl apply -f deploy/cephfs/kubernetes/v1.14+/csi-cephfsplugin.yaml
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+daemonset.apps/csi-cephfsplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+service/csi-metrics-cephfsplugin configured
+```
+
```

Comment:
Should we provide any commands to check the state of the daemon set, to see if the new pods have started? For example like [this](https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/#step-4-watching-the-rolling-update-status).

Reply:
am not sure how  to check daemonset state  if the update strategy  is OnDelete, we can revisit this section as  we add more things to the upgrade doc